src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -6940,6 +6941,133 @@ def merge(
+        allow_exact_matches : bool, default True
+            If True, allow matching with the same ‘on’ value (i.e. less-than-or-equal-to / greater-than-or-equal-to)
+            If False, don’t match the same ‘on’ value (i.e., strictly less-than / strictly greater-than).
+        direction : ‘backward’ (default), ‘forward’, or ‘nearest’
+            Whether to search for prior, subsequent, or closest matches.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler
+        """
+        if (
+            by
+            or left_by
+            or right_by
+            or left_index
+            or right_index
+            or tolerance
+            or suffixes != ("_x", "_y")
+        ):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas merge_asof method does not currently support parameters "
+                + "'by', 'left_by', 'right_by', 'left_index', 'right_index', "
+                + "'suffixes', or 'tolerance'"
+            )
+        if direction not in ("backward", "forward"):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas merge_asof method only supports directions 'forward' and 'backward'"
+            )
+
+        left_frame = self._modin_frame
+        right_frame = right._modin_frame
+        if on:
+            left_match_quoted_identifier = (
+                left_frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+                    [on]
+                )[0][0]
+            )
+            right_match_quoted_identifier = (
+                right_frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+                    [on]
+                )[0][0]
+            )
+        else:
+            assert left_on and right_on
+            left_match_quoted_identifier = (
+                left_frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+                    [left_on]
+                )[0][0]
+            )
+            right_match_quoted_identifier = (
+                right_frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+                    [right_on]
+                )[0][0]
+            )
+
+        if direction == "backward":
+            comparator_match_condition = "__ge__" if allow_exact_matches else "__gt__"
+        else:
+            comparator_match_condition = "__le__" if allow_exact_matches else "__lt__"
+
+        joined_frame, _ = join_utils.join(
+            left=left_frame,
+            right=right_frame,
+            how="asof",
+            left_on=[left_match_quoted_identifier],
+            right_on=[right_match_quoted_identifier],
+            left_match_condition=left_match_quoted_identifier,
+            right_match_condition=right_match_quoted_identifier,
+            comparator_match_condition=comparator_match_condition,
+            join_key_coalesce_config=[JoinKeyCoalesceConfig.LEFT] if on else None,
```

Comment:
this seems behaves the same as merge, if the left_on column name and right_on column is the same, the result column will have only one column, otherwise it will be the same column. For example, if left_on='a', right_on='a', the result will only have one a, other wise it will have two columns.

Reply:
you should be able to reuse the following function to get the join keys
```
left_keys, right_keys = join_utils.get_join_keys(
                left._modin_frame,
                right._modin_frame,
                on,
                left_on,
                right_on,
                left_index,
                right_index,
            )
```
and the same logic for decising the colease config used in merge can probably also be reused
```
 coalesce_config = []
        for lkey, rkey in zip(left_keys, right_keys):
            if lkey == rkey or rkey in external_join_keys:
                coalesce_config.append(join_utils.JoinKeyCoalesceConfig.LEFT)
            elif lkey in external_join_keys:
                coalesce_config.append(join_utils.JoinKeyCoalesceConfig.RIGHT)
            else:
                coalesce_config.append(join_utils.JoinKeyCoalesceConfig.NONE)
```
we can probably add another utilty function in join_utils to get the coalesce_config

Reply:
@sfc-gh-nkrishna i am little bit confused here, it seems the join_key_coalesce_config is still None if user is using left_on and right_on, and the result will contain both columns. 
You might be confused about the comment of colease config, it is not a logic inside get_join_keys, the get_join_keys helps figure out what are the correct join keys form left and right given all parameters related to key, but it doesn't figure out the coalesce_config, the coalesce_config logic I pasted above is actually from the merge configuration here https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py#L6876.
I was suggesting that we can extract this as a common util function, and call it for both merge, and this function here

Reply:
Ah that makes sense, thanks for the clarification!